src/dotnet-install.sh

```diff
@@ -905,10 +905,10 @@ install_dotnet() {
 
     # The download function will set variables $http_code and $download_error_msg in case of failure.
     download "$download_link" "$zip_path" 2>&1 || download_failed=true
-    primary_path_http_code="$http_code"; primary_path_download_error_msg="$download_error_msg"
 
     #  if the download fails, download the legacy_download_link
     if [ "$download_failed" = true ]; then
+        primary_path_http_code="$http_code"; primary_path_download_error_msg="$download_error_msg"
```

Comment:
This will still be unbound if you don't have wget or curl - download will fail, but you won't have an error code. @bozturkMSFT does it make sense to say, just exit the script if neither was found?

Reply:
That sounds reasonable enough to me at least so I updated the PR to do exactly that.

Reply:
Returning 1 from download() will sadly preserve this error behavior. Exiting directly from that function wont .

There's two options:
- Just log the message there and exit. There's little that can be done at any other point in the script that's useful to the users. This also avoids doing retries if there's no curl or wget.
- Change this to something like:
```suggestion
        primary_path_http_code="${http_code:-toolError}"; primary_path_download_error_msg="$download_error_msg"
```
The messages will repeat and do retries, but it's pretty small of a fix. 


Even with this contribution I'd be happy, as most error cases in the script will be "couldn't download", not "couldn't find something to download with", and this would fix issues that are pretty disruptive right now.